### PR TITLE
[DROOLS-6765] Update POI from 4.1.2 to 5.1.0

### PR DESCRIPTION
**Pull request by mariofusco**
**JIRA**: https://issues.redhat.com/browse/DROOLS-6765

Related PRs
https://github.com/kiegroup/drools/pull/4109
https://github.com/kiegroup/kogito-runtimes/pull/1845

### CODE REVIEW

File: kogito-maven-plugin/pom.xml [hunk 1]
```diff
@@ -63,6 +63,10 @@
           <groupId>org.eclipse.sisu</groupId>
           <artifactId>org.eclipse.sisu.plexus</artifactId>
         </exclusion>
+        <exclusion>
+          <groupId>commons-io</groupId>
+          <artifactId>commons-io</artifactId>
+        </exclusion>
```
Comment (danielezonca) [inline kogito-maven-plugin/pom.xml:69]: Can you please explain why this change and how this is related with the update of Apache POI?

Comment (mariofusco) [reply kogito-maven-plugin/pom.xml:69]: see https://github.com/kiegroup/drools/pull/4107#issuecomment-1008655945
